server/src/main/java/org/elasticsearch/common/geo/GeoUtils.java

```diff
@@ -505,23 +504,25 @@ public static GeoPoint parseGeoPoint(XContentParser parser, GeoPoint point, fina
      * @return int representing precision
      */
     public static int parsePrecision(XContentParser parser) throws IOException, ElasticsearchParseException {
-        XContentParser.Token token = parser.currentToken();
-        if (token.equals(XContentParser.Token.VALUE_NUMBER)) {
-            return XContentMapValues.nodeIntegerValue(parser.intValue());
-        } else {
-            String precision = parser.text();
+        return parser.currentToken() == Token.VALUE_NUMBER ? parser.intValue() : parsePrecisionString(parser.text());
+    }
+
+    /**
+     * Attempt to parse geohash precision string into an integer value
+     */
+    public static int parsePrecisionString(String precision) {
+        try {
+            // we want to treat simple integer strings as precision levels, not distances
+            return checkPrecisionRange(Integer.parseInt(precision));
```

Comment:
This is a breaking change here. We can only do this in 7.0.0 but I suspect you are going to want to backport this PR to the 6.x branch. I think we should change this here to warn if the precision is out of range using the deprecation logger and then we can have a follow up PR for 7.0 only that makes this strictly within the range.

Reply:
Its a breaking change because if the string in the JSON was a simple integer value before we would parse it to an int using `parser.intValue()` and not perform a range check. Now we do perform a range check so JSON that used to be parsed without error will now throw an exception, hence its a breaking change. I do agree that throwing an error is better but we need to be careful because this can make upgrades tricky for the user if suddenly requests that they used fine in their app before start failing.

The question is, what used to happen is you gave a precision value outside of the range? Did we fail later in execution (in which case the breaking change here might be ok)? Did we accept the precision value and process it at the  precision specified even though its outside the range? or may we accepted the precision value and if it was outside the range we used it as if it was at one of the bounds (i.e. if the value is above the MAX_PRECISION we just evaluated it as if it was MAX_PRECISION)?

Reply:
Ok, in which case I agree that this is not a breaking change